### DISCUSSION TITLE
enable/disable firewall rule

### DISCUSSION THREAD

**Discussion by xtiang7c0** [Art-of-WiFi/UniFi-API-client#175] [General]
Hi everyone,

thank you for the work you put into this.  I'm looking for a way to turn on/off firewall rules. via the API client but could not find a way with the commands in the documentation. Did I miss something?

thank you

**Comment by Brovning** [comment]
Hello,

I'm trying the same with PHP:
https://community.ui.com/questions/Activation-deactivation-of-PortForwarding-rules-in-UDM-via-curl-PHP/c321e5da-f2aa-4925-b94b-f1d04bd342c8

Unfortunately my code is not activating or deactivating the existing rules. With each execution it is creating additional port forwarding rules...

Any ideas on this?

**Comment by malle-pietje** [comment]
Haven’t been able to look into this yet…

**Comment by brakedalen** [comment]
I finally managed to get this connected to my firewall. However, I've realized that the functionality I require isn’t available here, despite ChatGPT indicating otherwise. Does anyone know how to enable and disable firewall rules through the API?

**Comment by malle-pietje** [comment]
Has anybody got a specific use case what needs to be achieved together with the code you’ve used so far?

**Comment by alexanderharm** [comment]
hi, i developed an app for nextcloud which does exactly that. i allows you to define some rules which can then be switch on/off with e. g. a timer. we use that at our local school so that teachers can allow/block access to the internet as needed. if anyone is interested i can publish the app...

**Reply by brakedalen** [reply]
That's very interesting! Are you using the standard API? I would to get this to work. The use case is enable/disable internet for my kids by using smart home controller like homey😉

**Reply by alexanderharm** [reply]
yes i do, it is in principle very simple... here is a working example using curl:

```
ADDRESS=10.10.10.1
USERNAME=myuser
PASSWORD=supersecret
RULEID=yhi37f7zeqe7bep6brbi6yjd

#login
csrftoken="$(curl -s -k -X POST --data "{\"username\": \"${USERNAME}\", \"password\": \"${PASSWORD}\"}" --header 'Content-Type: application/json' -c cookie.txt "https://${ADDRESS}/api/auth/login" -o /dev/null -D - | grep x-csrf-token)"

#enable
curl -s -k -X PUT -H "${csrftoken}" -b cookie.txt --data '{"enabled": true}' --header 'Content-Type: application/json' "https://${ADDRESS}/proxy/network/api/s/default/rest/firewallrule/${RULEID}" -o /dev/null

#disable
curl -s -k -X PUT -H "${csrftoken}" -b cookie.txt --data '{"enabled": false}' --header 'Content-Type: application/json' "https://${ADDRESS}/proxy/network/api/s/default/rest/firewallrule/${RULEID}" -o /dev/null

#logout
curl -s -k -X POST -H "${csrftoken}" -b cookie.txt "https://${ADDRESS}/api/logout" -o /dev/null
```

i can also provide this in php... but you get the idea...

**Reply by alexanderharm** [reply]
you can get all rules (to extract the id) by executing:

```
curl -s -k -X GET -b cookie.txt -H "${csrftoken}" "https://${ADDRESS}/proxy/network/api/s/default/rest/firewallrule"
```

which you could pipe to `jq` to improve readability or apply filters

**Comment by malle-pietje** [comment]
@alexanderharm thanks for sharing!

For those looking to leverage these suggestions through the API Client you can use the `custom_api_request()` method here:
https://github.com/Art-of-WiFi/UniFi-API-client/blob/main/src/Client.php#L3771-L3806

Make sure you use the correct URL, payload and HTTP method.
